core/src/main/java/uk/gov/justice/services/core/interceptor/InterceptorObserver.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.services.core.interceptor;
+
+import uk.gov.justice.services.core.annotation.AnyLiteral;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.Extension;
+
+/**
+ * Observes for {@link AfterDeploymentValidation} and adds all {@link Interceptor} implementations
+ * to the {@link InterceptorCache}
+ */
+public class InterceptorObserver implements Extension {
```

Comment:
If you are happy with the current implementation, can we merge for now and discuss the best practice for this, next week